setting up alternative project

ahcyip opened the pull request:
- separate dataset dimension_mapping into old /project and new /project_transportonly
- new project.json5 and /dimensions in new /project_transportonly

ahcyip committed:
setting up alternative project

- separate dataset dimension_mapping into old /project and new /project_transportonly
- new project.json5 and /dimensions in new /project_transportonly

ahcyip commented:
Hey Dan, this is to facilitate the publishing of stand-alone transportation data. I added a few more slices, added the required lookups/mappings, and set it up in the new structure as discussed (separate project folders, project.json5, and separate dataset/modeled/transport/project/dimension_mapping  &  dataset/modeled/transport/project_transportonly/dimension_mapping
![image](https://github.com/user-attachments/assets/bd07fce2-5933-436a-bdca-6dfedb82d806)

daniel-thom commented on project_transportonly/dimensions/subset/subsectors_by_sector.csv:
```diff
@@ -0,0 +1,25 @@
+id,commercial_subsectors,industrial_subsectors,residential_subsectors,transportation_subsectors
```
This file needs some work. The subsector IDs need to match the project's base dimension subsectors (like `Bus+Local+BEV_200`).

daniel-thom replied:
industrial, commercial, etc., can be deleted.

daniel-thom commented on project_transportonly/dimensions/subset/end_uses_by_sector.csv:
```diff
@@ -0,0 +1,40 @@
+﻿id,residential_end_uses,commercial_end_uses,transportation_end_uses,industrial_end_uses
+electricity_pools_and_spas,x,,,
```
Non-transportation end uses need to be deleted.

daniel-thom commented on project_transportonly/project.json5:
```diff
@@ -0,0 +1,354 @@
+                name: 'US Counties 2020 L48',
+                display_name: 'County',
+                description: 'US counties for L48, Census 2020 vintage, no territories',
+                file: 'dimensions/counties.csv',
```
This will likely eventually be very problematic. Counties come and go. We should strive for a minimal number of lists. This records will get de-duplicated at registration time. However, if a county gets added or removed, this copy may not get updated and will cause a hard-to-find bug. Can you point to the file paths in the main project for all duplicate record files?

ahcyip replied:
yes. I also pointed all other duplicates, dimensions, supplemental dimensions and supplemental dimension mappings, to ../project/xyz
I also added a supplemental dimension for the simplified vehicle types.

daniel-thom commented on project_transportonly/project.json5:
```diff
@@ -0,0 +1,354 @@
+                create_supplemental_dimension: false,
+                selectors: [
+                    {
+                        name: 'residential_end_uses',
```
Remove all non-transport items.

daniel-thom commented on project_transportonly/project.json5:
```diff
@@ -0,0 +1,354 @@
+                create_supplemental_dimension: false,
+                selectors: [
+                    {
+                        name: 'residential_subsectors',
```
Here too

ahcyip committed:
eliminate duplicate lists, add supplemental dimension of simplified vehicle type

now points to original project for all other duplicates, dimensions, supplemental dimensions and supplemental dimension mappings, to ../project/xyz

ahcyip commented:
Thanks @daniel-thom - how's this?

daniel-thom commented on project_transportonly/project.json5:
```diff
@@ -0,0 +1,340 @@
+                mapping: {
+                    description: 'Maps Detailed Vehicle Types to Simplified Vehicle Types',
+                    file: 'dimension_mappings/lookup_subsector_to_simplified_vehicle_type.csv',
+                    mapping_type: 'many_to_one_aggregation',
```
Slight error here. Need a closing brace for mapping.

daniel-thom commented on project_transportonly/project.json5:
```diff
@@ -0,0 +1,340 @@
+        supplemental_dimensions: [
+            {
+                type: 'subsector',
+				class: 'Simplified Vehicle Type',
```
This causes an error because the class doesn’t exist. It should probably just be `Subsector`.

daniel-thom commented on project_transportonly/project.json5:
```diff
@@ -0,0 +1,340 @@
+                name: 'Sectors - Transportation Only',
+                display_name: 'Sector',
+                description: 'DECARB 2023 Demand Sector; Transportation',
+                file: 'dimensions/sectors.csv',
```
This is the wrong path.

ahcyip replied:
accidentally deleted too many files... now back.

ahcyip committed:
bug fixes

ahcyip committed:
agg PHEVs, eliminate subset sectors, and 1 exception to MHDV

MHDV exception: base_missing 11001 county (DC) doesn't have MHDV light or heavy regional trucks (but does have medium regional)

ahcyip committed:
Update dimension_mappings.json5

fix many_to_one_aggregation specification

ahcyip committed:
Update project.json5

removing 4 more impossible MHDV combinations in 11001 DC

daniel-thom reviewed (approved):
I was able to register the project and dimensions. Any comments @elainethale?

ahcyip committed:
further minimizing duplication

referring to original project's dataset's mappings where possible (since they are unchanged)

elainethale commented on project_transportonly/project.json5:
```diff
@@ -0,0 +1,351 @@
+            {
+                type: 'subsector',
+                class: 'Subsector',
+                name: 'DECARB 2023 Demand Subsectors',
```
```suggestion
                name: 'DECARB 2023 Transport Subsectors',
```

elainethale commented on project_transportonly/project.json5:
```diff
@@ -0,0 +1,351 @@
+                class: 'Subsector',
+                name: 'DECARB 2023 Demand Subsectors',
+                display_name: 'Subsector',
+                description: 'DECARB 2023 Demand Subsectors; Vehicle Class',
```
As above, "Transport" subsectors instead of "Demand". Also, is "Vehicle Class" sufficient to describe what's in the subsector ids/names, e.g., "Personal_LDV+Pickup+BEV_200"?

elainethale commented on project_transportonly/project.json5:
```diff
@@ -0,0 +1,351 @@
+            {
+                type: 'metric',
+                class: 'EnergyEndUse',
+                name: 'DECARB 2023 End Uses',
```
```suggestion
                name: 'DECARB 2023 Transport End Uses',
```

elainethale commented on project_transportonly/project.json5:
```diff
@@ -0,0 +1,351 @@
+                class: 'EnergyEndUse',
+                name: 'DECARB 2023 End Uses',
+                display_name: 'End Use',
+                description: 'DECARB 2023 Demand End Uses; Charging Electricity Types',
```
```suggestion
                description: 'DECARB 2023 Transport End Uses; Charging Locations and Levels plus Transit Rail',
```

elainethale commented on project_transportonly/dimensions/enduses.csv:
```diff
@@ -0,0 +1,13 @@
+electricity_ev_mhdv_depot_ac,ev_mhdv_depot_ac,electricity,MWh
+electricity_ev_mhdv_depot_dc,ev_mhdv_depot_dc,electricity,MWh
+electricity_ev_mhdv_opportunity_dc,ev_mhdv_opportunity_dc,electricity,MWh
+electricity_ev_mhdv_enroute_dc,ev_mhdv_enroute_dc,electricity,MWh
+electricity_rail_transit,Electric Transit Rail,electricity,MWh
```
Sorry I didn't catch this in the main project, but should EVs have pretty names like "Electric Transit Rail" does, e.g., "EV M/HDV Enroute DC"?

elainethale commented on project_transportonly/project.json5:
```diff
@@ -0,0 +1,351 @@
+            {
+                type: 'scenario',
+                class: 'Scenario',
+                name: 'DECARB 2023 High-Resolution Demand Scenarios',
```
```suggestion
                name: 'DECARB 2023 Transport Scenarios',
```

elainethale commented on project_transportonly/project.json5:
```diff
@@ -0,0 +1,351 @@
+                class: 'Scenario',
+                name: 'DECARB 2023 High-Resolution Demand Scenarios',
+                display_name: 'Scenario',
+                description: 'DECARB 2023 High-Resolution Demand Scenarios; DECARB High, with two charging behaviors: unmanaged and overnight slow',
```
```suggestion
                description: 'DECARB 2023 Transport Scenarios; DECARB High, with two charging behaviors: unmanaged and overnight slow',
```

elainethale commented on project_transportonly/project.json5:
```diff
@@ -0,0 +1,351 @@
+                module: 'dsgrid.dimension.standard',
+            },
+        ],
+        subset_dimensions: [
+
+        ],
```
nit: remove whitespace?

elainethale replied:
Also, a bigger comment that the simplified vehicle types could be specified as Subset Dimensions instead of as a supplemental dimension.

elainethale commented on project_transportonly/project.json5:
```diff
@@ -0,0 +1,351 @@
+        supplemental_dimensions: [
+            {
+                type: 'subsector',
+				class: 'Subsector',
```
nit: uneven indention

elainethale commented on project_transportonly/project.json5:
```diff
@@ -0,0 +1,351 @@
+                    description: 'Maps Detailed Vehicle Types to Simplified Vehicle Types',
+                    file: 'dimension_mappings/lookup_subsector_to_simplified_vehicle_type.csv',
+                    mapping_type: 'many_to_one_aggregation',
+				},
+            },
+			{
```
nit: Uneven indentation

elainethale commented on project_transportonly/project.json5:
```diff
@@ -0,0 +1,351 @@
+                    mapping_type: 'many_to_one_aggregation',
+               },
+            },
+            {
+                // For regional reporting - Census Division does not align well with
+                // EMM/eGrid/GEA regions, so only report groups of states at this much 
+                // higher level
+                type: 'geography',
+                class: 'CensusRegion',
+                name: 'Census Region',
+                display_name: 'Census Region',
+                description: 'US Census Regions',
+                file: '../project/dimensions/supplemental/census_regions.csv',
+                module: 'dsgrid.dimension.standard',
+                mapping: {
+                    description: 'Maps US Counties 2020 L48 to CensusRegion',
+                    file: '../project/dimension_mappings/lookup_county_to_censusregion.csv',
+                    mapping_type: 'many_to_one_aggregation',
+                },
```
We skip census division level for the main project because of the buildings dataset, but that concern doesn't apply here. Thus, you could add Census Division back in if you want.

elainethale commented on project_transportonly/project.json5:
```diff
@@ -0,0 +1,351 @@
+               },
+            },
+            {
+                // For regional reporting - Census Division does not align well with
+                // EMM/eGrid/GEA regions, so only report groups of states at this much 
+                // higher level
```
```suggestion
```

elainethale reviewed (changes requested):
Instead of `project_transportonly`, can we use the name `project_transport`? 

If you really want to keep "only" in the name, let's separate out the words: `project_transport_only`.

I'm explicitly requesting changes because the simplified vehicle types are inconsistent.

ahcyip committed:
Update datasets/modeled/transport/project_transportonly/dimension_mappings.json5

Co-authored-by: Elaine Hale <elaine.hale@nrel.gov>

ahcyip committed:
Update datasets/modeled/transport/project_transportonly/dimension_mappings.json5

Co-authored-by: Elaine Hale <elaine.hale@nrel.gov>

ahcyip committed:
Update datasets/modeled/transport/project_transportonly/dimension_mappings.json5

Co-authored-by: Elaine Hale <elaine.hale@nrel.gov>

ahcyip committed:
Update project_transportonly/dimensions/scenarios.csv

Co-authored-by: Elaine Hale <elaine.hale@nrel.gov>

ahcyip committed:
Update project_transportonly/project.json5

Co-authored-by: Elaine Hale <elaine.hale@nrel.gov>

ahcyip committed:
Update project_transportonly/project.json5

Co-authored-by: Elaine Hale <elaine.hale@nrel.gov>

ahcyip committed:
Update project_transportonly/dimensions/scenarios.csv

Co-authored-by: Elaine Hale <elaine.hale@nrel.gov>

ahcyip committed:
Apply suggestions from code review

Co-authored-by: Elaine Hale <elaine.hale@nrel.gov>

ahcyip committed:
rename to project_transport_only, changing model_years

 (project now reverted back to original transport dataset years)

ahcyip committed:
finalizing more detailed vehicle types, names, and descriptions, for both /project and /project_transport_only

ahcyip commented:
@elainethale I've made those suggested changes in both /project and /project_transport_only

@daniel-thom In addition to these labeling changes, I've also made a change to the model_year dimension for /project_transport_only (that follows straight from dataset, so no mapping needed). With this, could you please re-run all 3 slices (full_dataset, state_simplified, county_summary)?

daniel-thom commented on project/dimensions/subsectors.csv:
```diff
@@ -8,10 +8,10 @@ phev_compact,PHEV Compact Car
-bev_medium_truck,BEV Medium Freight Truck
-bev_non_freight_truck,BEV Non-Freight Truck
-bev_heavy_truck,BEV Heavy Truck
+bev_light-medium_freight_truck,BEV Light-Medium Freight Truck
```
Would you mind always using underscores instead of dashes? This inconsistency is bound to cause user problems.

elainethale replied:
I actually asked for the dash in this case, because it means "light-medium" and not "light and medium".

daniel-thom commented on project/dimensions/subset/subsectors_by_sector.csv:
```diff
@@ -8,10 +8,10 @@ phev_compact,,,,x
-bev_medium_truck,,,,x
-bev_non_freight_truck,,,,x
-bev_heavy_truck,,,,x
+bev_light-medium_freight_truck,,,,x
```
same here

ahcyip commented:
I noticed that we get a value of 2018 under a "year"? column when we do "annual time" - could we drop it instead? 

I'll also note that we have some confusion from the transport world where "Model Year" typically refers to vehicle model year (a 2020 Chevrolet Bolt). Would Simulation Year be better? Not strongly pushing if it's difficult or there are other reasons.

Here is the updated chart: 
![image](https://github.com/user-attachments/assets/1b592141-44cd-48cb-b216-2e1de5c3e54f)

elainethale commented on project_transport_only/dimensions/enduses.csv:
```diff
@@ -0,0 +1,13 @@
+id,name,fuel_id,unit
+electricity_ev_ldv_home_l1,Home Level 1 Light Duty EV Charging,electricity,MWh
+electricity_ev_ldv_home_l2,Home Level 2 Light Duty EV Charging,electricity,MWh
+electricity_ev_ldv_work_l1,Workplace Level 1 Light Duty EV Charging,electricity,MWh
+electricity_ev_ldv_work_l2,Workplace Level 2 Light Duty EV Charging,electricity,MWh
+electricity_ev_ldv_public_l2,Public Level 2 Light Duty EV Charging,electricity,MWh
+electricity_ev_ldv_public_dcfc,Public DCFC Light Duty EV Charging,electricity,MWh
+electricity_ev_ldv_enroute_dcfc,Enroute DCFC Light Duty EV Charging,electricity,MWh
+electricity_ev_mhdv_depot_ac,Depot AC Medium and Heavy Duty EV Charging,electricity,MWh
+electricity_ev_mhdv_depot_dc,Depot DC Medium and Heavy Duty EV Charging,electricity,MWh
+electricity_ev_mhdv_opportunity_dc,Opportunity DC Medium and Heavy Duty EV Charging,electricity,MWh
+electricity_ev_mhdv_enroute_dc,Enroute DC Medium and Heavy Duty EV Charging,electricity,MWh
```
Nit: Names could start from the bigger category and get more specific, e.g.:

```
Light-duty EV Home Level 1 Charging
Medium- and Heavy-duty EV Opportunity DC Charging
```

elainethale commented on project_transport_only/dimensions/supplemental/simplified_vehicle_types.csv:
```diff
@@ -0,0 +1,15 @@
+phev_suv,PHEV SUV
+bev_light-medium_freight_truck,BEV Light-Medium Freight Truck
+bev_medium_freight_truck,BEV Medium Freight Truck
+bev_vocational_truck,BEV Non-Freight Truck
```
```suggestion
bev_vocational_truck,BEV Vocational Truck
```

I think you use "BEV Vocational Truck" as the name under project? Maybe just copy-paste all those lines to be sure they're the same in both places?

elainethale commented on project_transport_only/project.json5:
```diff
@@ -0,0 +1,345 @@
+            {
+                type: 'scenario',
+                class: 'Scenario',
+                name: 'DECARB 2023 High-Resolution Demand Scenarios',
```
```suggestion
                name: 'DECARB 2023 High-Resolution Transportation Demand Scenarios',
```

elainethale commented on project_transport_only/project.json5:
```diff
@@ -0,0 +1,345 @@
+                class: 'Scenario',
+                name: 'DECARB 2023 High-Resolution Demand Scenarios',
+                display_name: 'Scenario',
+                description: 'DECARB 2023 High-Resolution Demand Scenarios; DECARB High, with two charging behaviors from TEMPO/EVIX/POLARIS modeling: Unmanaged and Overnight slow',
```
```suggestion
                description: 'DECARB 2023 High-Resolution Transportation Demand Scenarios; DECARB High, with two charging behaviors from TEMPO/EVIX/POLARIS modeling: Unmanaged and Overnight slow',
```

elainethale commented:
> I noticed that we get a value of 2018 under a "year"? column when we do "annual time" - could we drop it instead?
> 
> I'll also note that we have some confusion from the transport world where "Model Year" typically refers to vehicle model year (a 2020 Chevrolet Bolt). Would Simulation Year be better? Not strongly pushing if it's difficult or there are other reasons.
> 
> Here is the updated chart: ![image](https://private-user-images.githubusercontent.com/5713411/358148688-1b592141-44cd-48cb-b216-2e1de5c3e54f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM3NjM0NTgsIm5iZiI6MTcyMzc2MzE1OCwicGF0aCI6Ii81NzEzNDExLzM1ODE0ODY4OC0xYjU5MjE0MS00NGNkLTQ4Y2ItYjIxNi0yZTFkZTVjM2U1NGYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgxNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MTVUMjMwNTU4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OWFlZWM4Mzk3OGE3MzdhM2I1OWY0MzBkYjczN2VhZTBkN2RhMGJkODM1OGI4MGNiNDMwZWM0NTBhYmJmMDFlOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.V_Fr3YSh-2Eq_t2dhtHmi-XMzDs5NRa9UffDu41tWo0)

It seems like the annual time queries could drop the time column altogether. Is that correct @daniel-thom?

The "model_year" term is tougher. I understand the confusion for the transportation sector in particular, but it is a dsgrid term and leaving it as the column name is going to be the most consistent with other published datasets. ("simulation_year" also isn't always accurate for all the datasets in dsgrid, since not all the datasets are "simulated".)

daniel-thom commented:
> It seems like the annual time queries could drop the time column altogether. Is that correct @daniel-thom?

Yes. We can add a second aggregation to the chain that drops it. I can do that when we’re ready to re-run.

daniel-thom committed:
Add dimensions for electricity end uses

daniel-thom committed:
Remove tabs to fix alignment

daniel-thom committed:
Merge pull request #31 from dsgrid/dt/project-transport-only

Add subset dimension for end uses by fuel type

ahcyip committed:
Apply suggestions from code review

Co-authored-by: Elaine Hale <elaine.hale@nrel.gov>

ahcyip committed:
improved names for EV charging end uses as suggested by @elainethale

ahcyip committed:
adjust indenting

ahcyip commented:
More data incoming... but I'm going to merge here.
